doc/v5.0/man3/MPI_File_seek.3
.\" -*- nroff -*-
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright 2015-2016 Research Organization for Information Science
.\"                     and Technology (RIST). All rights reserved.
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_File_seek 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_File_seek\fP \- Updates individual file pointers (noncollective).

.SH SYNTAX
.ft R
.nf
.SH C Syntax
.nf
#include <mpi.h>
int MPI_File_seek(MPI_File \fIfh\fP, MPI_Offset \fIoffset\fP,
	int \fIwhence\fP)

.fi
.SH Fortran Syntax (see FORTRAN 77 NOTES)
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_FILE_SEEK(\fIFH\fP, \fIOFFSET\fP, \fIWHENCE\fP, \fIIERROR\fP)
	INTEGER	\fIFH, WHENCE, IERROR\fP
	INTEGER(KIND=MPI_OFFSET_KIND)	\fIOFFSET\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_File_seek(\fIfh\fP, \fIoffset\fP, \fIwhence\fP, \fIierror\fP)
	TYPE(MPI_File), INTENT(IN) :: \fIfh\fP
	INTEGER(KIND=MPI_OFFSET_KIND), INTENT(IN) :: \fIoffset\fP
	INTEGER, INTENT(IN) :: \fIwhence\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT PARAMETERS
.ft R
.TP 1i
fh
File handle (handle).
.TP 1i
offset
File offset (integer).
.TP 1i
whence
Update mode (integer).

.SH OUTPUT PARAMETER
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
MPI_File_seek updates the individual file pointer according to
.I whence,
which could have the following possible values:
.TP
  o
MPI_SEEK_SET - The pointer is set to
.I offset.
.TP
  o
MPI_SEEK_CUR - The pointer is set to the current pointer position plus
.I offset.
.TP
  o
MPI_SEEK_END - The pointer is set to the end of the file plus
.I offset.
.sp
.RE
The
.I offset
can be negative, which allows seeking backwards. It is erroneous to
seek to a negative position in the file. The end of the file is
defined to be the location of the next elementary data item
immediately after the last accessed data item, even if that location
is a hole.

.SH FORTRAN 77 NOTES
.ft R
The MPI standard prescribes portable Fortran syntax for
the \fIOFFSET\fP argument only for Fortran 90.  FORTRAN 77
users may use the non-portable syntax
.sp
.nf
     INTEGER*MPI_OFFSET_KIND \fIOFFSET\fP
.fi
.sp
where MPI_OFFSET_KIND is a constant defined in mpif.h
and gives the length of the declared integer in bytes.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. For MPI I/O function errors, the default error handler is set to MPI_ERRORS_RETURN. The error handler may be changed with MPI_File_set_errhandler; the predefined error handler MPI_ERRORS_ARE_FATAL may be used to make I/O errors fatal. Note that MPI does not guarantee that an MPI program can continue past an error.



doc/v1.4/man3/MPI_File_seek.3.php
<?php
$topdir = "../../..";
$title = "MPI_File_seek(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_FILE_SEEK(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_File_seek</b> - Updates individual file pointers (noncollective).
<p>

<h2><a name='sect1' href='#toc1'>Syntax</a></h2>
<br>
<pre>C Syntax
    #include &lt;mpi.h&gt;
    int MPI_File_seek(MPI_File fh, MPI_Offset offset,
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      <tt> </tt>&nbsp;<tt> </tt>&nbsp;   int whence)
</pre>
<h2><a name='sect2' href='#toc2'>Fortran Syntax (see FORTRAN 77 NOTES)</a></h2>
<br>
<pre>    INCLUDE &rsquo;mpif.h&rsquo;
    MPI_FILE_SEEK(FH, OFFSET, WHENCE, IERROR)
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      INTEGER FH, WHENCE, IERROR
    <tt> </tt>&nbsp;<tt> </tt>&nbsp;      INTEGER(KIND=MPI_OFFSET_KIND) OFFSET
</pre>
<h2><a name='sect3' href='#toc3'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::File::Seek(MPI::Offset offset, int whence)
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameters</a></h2>

<dl>

<dt>fh     </dt>
<dd>File handle (handle). </dd>

<dt>offset  </dt>
<dd>File offset (integer).
</dd>

<dt>whence </dt>
<dd>Update mode (integer).
<p> </dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameter</a></h2>

<dl>

<dt>IERROR </dt>
<dd>Fortran only: Error
status (integer).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
MPI_File_seek updates the individual file
pointer according to  <i>whence,</i> which could have the following possible values:
<br>

<dl>

<dt>  o   </dt>
<dd>MPI_SEEK_SET - The pointer is set to  <i>offset.</i> <br>
</dd>

<dt>  o   </dt>
<dd>MPI_SEEK_CUR - The pointer is set to the current pointer position
plus  <i>offset.</i> <br>
</dd>

<dt>  o   </dt>
<dd>MPI_SEEK_END - The pointer is set to the end of the file plus  <i>offset.</i>
<p>
</dd>
</dl>
</blockquote>
The  <i>offset</i>  can be negative, which allows seeking backwards. It is erroneous
to seek to a negative position in the file. The end of the file is defined
to be the location of the next elementary data item immediately after the
last accessed data item, even if that location is a hole.
<p>
<h2><a name='sect7' href='#toc7'>Fortran 77 Notes</a></h2>
The
MPI standard prescribes portable Fortran syntax for the <i>OFFSET</i> argument
only for Fortran 90.  FORTRAN 77 users may use the non-portable syntax <p>
<br>
<pre>     INTEGER*MPI_OFFSET_KIND OFFSET
</pre><p>
where MPI_OFFSET_KIND is a constant defined in mpif.h and gives the length
of the declared integer in bytes.
<p>
<h2><a name='sect8' href='#toc8'>Errors</a></h2>
Almost all MPI routines return
an error value; C routines as the value of the function and Fortran routines
in the last argument. C++ functions do not return errors. If the default
error handler is set to MPI::ERRORS_THROW_EXCEPTIONS, then on error the
C++ exception mechanism will be used to throw an MPI:Exception object. <p>
Before
the error value is returned, the current MPI error handler is called. For
MPI I/O function errors, the default error handler is set to MPI_ERRORS_RETURN.
The error handler may be changed with <a href="../man3/MPI_File_set_errhandler.3.php">MPI_File_set_errhandler</a>; the predefined
error handler MPI_ERRORS_ARE_FATAL may be used to make I/O errors fatal.
Note that MPI does not guarantee that an MPI program can continue past
an error.
<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>Fortran Syntax (see FORTRAN 77 NOTES)</a></li>
<li><a name='toc3' href='#sect3'>C++ Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameters</a></li>
<li><a name='toc5' href='#sect5'>Output Parameter</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Fortran 77 Notes</a></li>
<li><a name='toc8' href='#sect8'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
